tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/process/computer/TinkerGraphComputer.java

```diff
@@ -54,6 +56,12 @@
  */
 public final class TinkerGraphComputer implements GraphComputer {
 
+    static {
+        // GraphFilters are expensive w/ TinkerGraphComputer as everything is already in memory
```

Comment:
`TinkerGraphComputer` already has the entire graph loaded in memory. Thus, when you add a `GraphFilter`, what I'm doing is creating a `Set<Object>` of edge IDs that are allowed. Then, every `Vertex.edges()` is filtered by `edge.id() in Set<Object>`. Its lame. As such, edge-based `GraphFilters` are expensive in `TinkerGraphComputer`.